Functions/Initialization/xASL_init_LoadDataParameterFile.m

```diff
@@ -18,91 +16,42 @@
 %
 % -----------------------------------------------------------------------------------------------------------------------------------------------------
 % REFERENCES:  n/a
-%
-% Copyright 2015-2021 ExploreASL
-
+% __________________________________
+% Copyright (c) 2015-2022 ExploreASL
 
-    if SelectParFile
-        fprintf('ExploreASL requires a data parameter file, which can be either a .m or .json file\n');
-        DataParPath = input('Please provide the full path of the DataParameter file, including ": ');
-    end
 
-    [pathstr, ~, Dext] = fileparts(DataParPath);
+    %% Get file extension and back up the x structure
+    [pathstr, ~, Dext] = fileparts(x.dir.dataPar);
     xBackup = x;
 
-    if strcmp(Dext,'.mat') % Now reading a mat-file, soon to be .json table complying with BIDS
-        %% Mat file can only contain single variable, that should contain the x/parameters
-        TempVar = load(DataParPath);
-        FieldN = fields(TempVar);
-        x = TempVar.(FieldN{1});
-    elseif strcmp(Dext,'.json')
-        % JSON import
-        try
-            x = xASL_io_ReadDataPar(DataParPath, false);
-        catch ME1
-            % if this fails, try to recreate the json file from an .m file,
-            % if it exists
-            [Fpath, Ffile] = fileparts(DataParPath);
-            DataParPath = fullfile(Fpath, [Ffile '.m']);
-            if exist(DataParPath, 'file')
-                warning('Invalid DataPar JSON file, trying to repair from detected .m file');
-                fprintf('A common issue is needing escaping e.g. "\\d" instead of "\d"\n');
-                try
-                    x = xASL_io_ReadDataPar(DataParPath, false);
-                catch ME2
-                    fprintf('%s\n', ME1.message);
-                    fprintf('%s\n', ME2.message);
-                    fprintf('A common issue is needing escaping e.g. "\\d" instead of "\d"\n');
-                    error('Something went wrong loading the DataParFile');
-                end
-            else
-                warning('Invalid DataPar JSON file');
```

Comment:
I agree to remove the mat-files.
@HenkMutsaerts we shouldn't be needing those - these are really ancient, right?!

However, we might keep the code in a Development/Outdated function as a fallback if somebody has these data. So then the code here will say `wrong, m-file not supported, use DICOM` - like you have it. But it would also write what to do in case you would want to convert your files.

Reply:
I really don't want to point out the obvious, but wouldn't that person just use the old version? You could start to lift some restrictions regarding compatibility to pump up the progress speed.

Reply:
Improve the warning?